src/serializer/functions.js

```diff
@@ -135,6 +135,24 @@ export class Functions {
     applyOptimizedReactComponents(this.realm, this.writeEffects, environmentRecordIdAfterGlobalCode);
   }
 
+  // When __optimize calls are called, an optional "pure" property can be passed
+  // on a config object to disable pure scope, otherwise pure scope is enabled by default.
+  // i.e. __optimimze(foo, { pure: false });
+  _isPureOptimizedFunction(funcValue: FunctionValue): boolean {
+    let possibleConfig = this.realm.optimizedFunctionConfig.get(funcValue);
+
+    if (possibleConfig instanceof ObjectValue) {
+      let pureVal = Get(this.realm, possibleConfig, "pure");
```

Comment:
Rather use possibleConfig._SafeGetDataPropertyValue("pure"). While you are at, perhaps fix all of the other unsafe uses of Get in this file?